app/models/optimization.rb

```diff
@@ -154,4 +154,18 @@ def log_error(err_msg)
   def raise_error(err_msg)
     raise OptimizationError.new(err_msg)
   end
+
+  def nb_points
+    return self.inputs.empty? ? "0" : self.inputs["x"].length
```

Comment:
Hi Nicolas. You can remove `return`. In Ruby, the `return` keyword is optional when returning at the end of a function and in practice is only used when returning earlier.

Reply:
Yes, I forgot, thank you for reminding me, I will fix this.